src/slurmutils/core/base.py

```diff
@@ -396,7 +396,7 @@ def make_model_builder(
     def match(obj: Any) -> Model | ModelList | ModelMapping | None:
         for model in models:
             try:
-                validate(obj, schema=model.__model_schema__)
+                validate(obj, schema=cast(dict[str, Any], model.__model_schema__))
```

Comment:
Huh, interesting that pyright cannot deduce this type. Maybe it's because `__model_schema__` takes `self` as an argument, but is a `@classproperty` at the same time?